executor/aggfuncs/func_count_distinct.go

```diff
@@ -401,6 +405,7 @@ func evalAndEncode(
 		if err != nil || isNull {
 			break
 		}
+		val = string(collator.Key(val))
 		encodedBytes = codec.EncodeCompactBytes(encodedBytes, hack.Slice(val))
```

Comment:
codec.EncodeCompactBytes(encodedBytes, collator.Key(val))

Reply:
No need to bytes -> string -> bytes

Reply:
OK, updated.